sdk/core/azure-mgmt-core/azure/mgmt/core/policies/_authentication.py

```diff
@@ -62,6 +64,87 @@ def on_challenge(self, request, response):  # pylint:disable=unused-argument
         return False
 
 
+# pylint:disable=too-few-public-methods
+class _AuxiliaryAuthenticationPolicyBase(object):
+    """Adds auxiliary authorization token header to requests.
+
+    :param ~azure.core.credentials.TokenCredential auxiliary_credentials: auxiliary credential for authorizing requests
+    :param str scopes: required authentication scopes
+    """
+
+    def __init__(
+        self, auxiliary_credentials, *scopes, **kwargs
+    ):  # pylint: disable=unused-argument
+        super().__init__()
+        self._auxiliary_credentials = auxiliary_credentials
+        self._scopes = scopes
+        self._aux_tokens = None
+
+    @staticmethod
+    def _enforce_https(request):
+        # type: (PipelineRequest) -> None
+
+        # move 'enforce_https' from options to context, so it persists
+        # across retries but isn't passed to transport implementation
+        option = request.context.options.pop("enforce_https", None)
+
+        # True is the default setting; we needn't preserve an explicit opt in to the default behavior
+        if option is False:
+            request.context["enforce_https"] = option
+
+        enforce_https = request.context.get("enforce_https", True)
+        if enforce_https and not request.http_request.url.lower().startswith("https"):
+            raise ServiceRequestError(
+                "Bearer token authentication is not permitted for non-TLS protected (non-https) URLs."
+            )
+
+    def _update_headers(self, headers):
+        """Updates the Authorization header with the auxiliary token.
+
+        :param dict headers: The HTTP Request headers
+        """
+        if self._aux_tokens:
+            headers["x-ms-authorization-auxiliary"] = ", ".join(
+                "Bearer {}".format(token.token) for token in self._aux_tokens
+            )
+
+    @property
+    def _need_new_aux_tokens(self):
+        if not self._aux_tokens:
+            return True
+        for token in self._aux_tokens:
+            if token.expires_on - time.time() < 300:
+                return True
+        return False
+
+
+class AuxiliaryAuthenticationPolicy(
+    _AuxiliaryAuthenticationPolicyBase, SansIOHTTPPolicy
```

Comment:
```suggestion
    _AuxiliaryAuthenticationPolicyBase, SansIOHTTPPolicy[HTTPRequestTypeVar, HTTPResponseTypeVar]
```

And add the correct TypeVar at the top
```python
HTTPResponseTypeVar = TypeVar("HTTPResponseTypeVar")
HTTPRequestTypeVar = TypeVar("HTTPRequestTypeVar")
```